src/main/java/networkbook/logic/parser/ParserUtil.java

```diff
@@ -96,6 +100,37 @@ public static Email parseEmail(String email) throws ParseException {
         return new Email(trimmedEmail);
     }
 
+    /**
+     * Parses a {@code Collection<String>} of emails into {@code UniqueList<Email>}.
+     * Leading and trailing whitespaces will be trimmed.
+     * @throws ParseException if at least one email in {@code Collection<String>} is invalid.
+     */
+    public static UniqueList<Email> parseEmails(Collection<String> emails) throws ParseException {
+        requireNonNull(emails);
+
+        if (!verifyNoDuplicates(emails)) {
+            throw new ParseException(MESSAGE_EMAIL_DUPLICATE);
+        }
+
+        UniqueList<Email> result = new UniqueList<>();
+        for (String email: emails) {
+            result.add(parseEmail(email));
+        }
+        return result;
+    }
+
+    private static boolean verifyNoDuplicates(Collection<String> strings) {
+        Object[] stringArr = strings.toArray();
+        for (int i = 0; i < stringArr.length; i++) {
+            for (int j = i + 1; j < stringArr.length; j++) {
+                if (stringArr[i].equals(stringArr[j])) {
+                    return false;
```

Comment:
Do test the verifyNoDuplicates function on a Collection<String> instance containing duplicate elements!

Reply:
Done!